src/Appium.Net/Appium/Windows/WindowsOptions.cs

```diff
@@ -0,0 +1,185 @@
+        /// </summary>
+        public string AppTopLevelWindow { get; set; }
+
+        /// <summary>
+        /// Full path to the folder.
+        /// <remarks>This is only applicable for classic apps</remarks>
+        /// </summary>
+        public string AppWorkingDir { get; set; }
+
+        /// <summary>
+        /// Timeout in milliseconds used to retry Appium Windows Driver session startup.
+        /// <remarks>Default value is 20000</remarks>
+        /// </summary>
+        public int CreateSessionTimeout { get; set; } = 20000;
+
+        /// <summary>
+        /// Similar to <see cref="CreateSessionTimeout"/>, but in seconds and is applied on the server side.
+        /// <remarks>Default value is 50 seconds.</remarks>
+        /// <remarks>The limit for this is 50 seconds.</remarks>
+        /// </summary>
+        public int WaitForAppLaunchTimeout { get; set; } = 50;
+
+        /// <summary>
+        /// Enables experimental features and optimizations.
+        /// <remarks>This is <see langword="false"/> by default</remarks>
+        /// </summary>
+        public bool IsExperimentalWebDriver { get; set; }
+
+        /// <summary>
+        /// The port number to execute Appium Windows Driver server listener on.
+        /// <remarks>The default starting port number for a new Appium Windows Driver session is 4724.
+        /// If this port is already busy then the next free port will be automatically selected.</remarks>
+        /// </summary>
+        public int SystemPort { get; set; } = 4724;
+
+        /// <summary>
+        /// An object containing either script or command key.
+        /// The value of each key must be a valid PowerShell script or command to be executed prior to the WinAppDriver session startup.
+        /// Example: <c>script: 'Get-Process outlook -ErrorAction SilentlyContinue'}</c>
+        /// </summary>
+        public string PreRunScript { get; set; }
+
+        /// <summary>
+        ///An object containing either script or command key.
+        /// The value of each key must be a valid PowerShell script or command to be executed after WinAppDriver session is stopped.
+        /// </summary>
+        public string PostRunScript { get; set; }
+
+
+        [Obsolete(
+            "Use the temporary AddAdditionalOption method or the browser-specific method for adding additional options")]
+        public override void AddAdditionalCapability(string capabilityName, object capabilityValue)
+        {
+            if (IsNullOrEmpty(capabilityName))
+            {
+                throw new ArgumentException("Capability name may not be null an empty string.", "capabilityName");
+            }
+
+            AddAdditionalOption(capabilityName, capabilityValue);
+        }
+
+        public override ICapabilities ToCapabilities()
+        {
+            var capabilities = GenerateDesiredCapabilities(false);
+
+            if (!IsNullOrEmpty(PlatformName))
```

Comment:
can we avoid code duplication here?

Reply:
Yeah I'm going to call the base method, I just realised not all caps were being picked up because of that